consul/src/main/scala/io/buoyant/consul/CatalogNamer.scala

```diff
@@ -2,16 +2,30 @@ package io.buoyant.consul
 
 import com.twitter.finagle._
 import com.twitter.util._
-import io.buoyant.consul.v1.{UnexpectedResponse, ServiceNode}
+import io.buoyant.consul.v1.{ServiceNode, UnexpectedResponse}
+import io.buoyant.namer.Metadata
 
 class CatalogNamer(
   idPrefix: Path,
-  api: v1.CatalogApi,
-  includeTag: Boolean = false
+  catalogApi: v1.CatalogApi,
+  agentApi: v1.AgentApi,
+  includeTag: Boolean = false,
+  setHost: Boolean = false
 ) extends Namer {
 
   import CatalogNamer._
 
+  private[this] lazy val domainFuture: Future[String] =
+    agentApi.self().flatMap { self =>
+      val domain = self.Config
+        .flatMap(_.Domain)
+        .map(_.stripPrefix(".").stripSuffix("."))
+      domain match {
+        case Some(d) => Future.value(d)
+        case None => Future.exception(???)
```

Comment:
I asked about that in a comment above. :)

I see 2 options:
1) Some concrete exception - in that case can you recommend one? :)
2) Rather than exception fallback to `consul` - the default value according to Consul docs.

What would you prefer?